deepcell/datasets/builders.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2016-2021 The Van Valen Lab at the California Institute of
+# Technology (Caltech), with support from the Paul Allen Family Foundation,
+# Google, & National Institutes of Health (NIH) under Grant U24CA224309-01.
+# All rights reserved.
+#
+# Licensed under a modified Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.github.com/vanvalenlab/deepcell-tf/LICENSE
+#
+# The Work provided may be used for non-commercial academic purposes only.
+# For any other use of the Work, including commercial use, please contact:
+# vanvalenlab@gmail.com
+#
+# Neither the name of Caltech nor the names of its contributors may be used
+# to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Dataset Builders"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_addons as tfa
+
+
+class TrackingDatasetBuilder(object):
+    """Create a tf.data.Dataset object for training tracking models.
+
+    Args:
+        track_list (list): A list of `track` objects containing lineage info
+        track_length (int): Number of desired frames per track
+        batch_size (int): Number of desired batches per dataset
+    """
+    def __init__(self,
+                 track_list,
+                 track_length=8,
+                 batch_size=1):
+        self.batch_size = batch_size
+        self.track_list = track_list
+        self.track_length = track_length
+
+        # Create max_nodes and max_frames
+        self._create_max_sizes()
+
+        # Load tracks
+        self._load_tracks()
+
+        # Create datasets
+        self._create_dataset()
```

Comment:
Not a fan of the pattern, setting attributes in helper functions rather than in `__init__`.

Reply:
Corrected for clarity.